tests/unit/utils/data.test.ts

```diff
@@ -17,9 +17,27 @@ const uint8ArraysAreEqual = (a: Uint8Array, b: Uint8Array) => {
   return a.every((val, i) => val === b[i]);
 };
 
+const uint8ArrayAndArrayBufferAreEqual = (
+  uint8Array: Uint8Array,
+  buffer: ArrayBuffer
+): boolean => {
+  const bufferUint8Array = new Uint8Array(buffer);
+  if (bufferUint8Array.length !== uint8Array.length) {
+    return false;
+  }
+  for (let i = 0; i < bufferUint8Array.length; i++) {
+    if (bufferUint8Array[i] !== uint8Array[i]) {
+      return false;
+    }
+  }
+  return true;
+};
+
 let data: any;
```

Comment:
node `fs/promises` `readFile()` returns a `Buffer`, for the sake of testing with `ArrayBuffer` we need to convert node's `Buffer` to an `ArrayBuffer`

```suggestion
const nodeBufferToArrayBuffer = (nodeBuffer: Buffer): ArrayBuffer =>
  nodeBuffer.buffer.slice(
    nodeBuffer.byteOffset,
    nodeBuffer.byteLength + nodeBuffer.byteOffset
  );


let data: any;
```